hail/python/hail/methods/relatedness/identity_by_descent.py

```diff
@@ -87,18 +91,138 @@ def identity_by_descent(dataset, maf=None, bounded=True, min=None, max=None) ->
 
     require_col_key_str(dataset, 'identity_by_descent')
 
+    if not isinstance(dataset.GT, hl.CallExpression):
+        raise Exception('GT field must be of type Call')
+
     if maf is not None:
         analyze('identity_by_descent/maf', maf, dataset._row_indices)
         dataset = dataset.select_rows(__maf=maf)
+        dataset = dataset.filter_rows(hl.is_defined(dataset.__maf))
     else:
         dataset = dataset.select_rows()
+
     dataset = dataset.select_cols().select_globals().select_entries('GT')
     dataset = require_biallelic(dataset, 'ibd')
 
-    return Table(ir.MatrixToTableApply(dataset._mir, {
-        'name': 'IBD',
-        'mafFieldName': '__maf' if maf is not None else None,
-        'bounded': bounded,
-        'min': min,
-        'max': max,
-    })).persist()
+    if isinstance(Env.backend(), SparkBackend):
+        return Table(ir.MatrixToTableApply(dataset._mir, {
+            'name': 'IBD',
+            'mafFieldName': '__maf' if maf is not None else None,
+            'bounded': bounded,
+            'min': min,
+            'max': max,
+        })).persist()
+
+    min = min or 0
+    max = max or 1
+    if not 0 <= min <= max <= 1:
+        raise Exception(f"invalid pi hat filters {min} {max}")
+
+    sample_ids = dataset.s.collect()
```

Comment:
Since we're making multiple passes over `dataset`, we probably want to checkpoint it before this first pass.